Source/DafnyCore/Verifier/ReadFrameSubset.cs

```diff
@@ -27,8 +27,18 @@ List<string> GetMessageParts() {
         obj = Printer.ExprToString(DafnyOptions.DefaultImmutableOptions, m.Array,
           new PrintFlags(UseOriginalDafnyNames: true));
       }
-
-      if (scope is Function { CoClusterTarget: var x } && x != Function.CoCallClusterInvolvement.None) {
+      // TODO is this a good error message for invariants?
```

Comment:
I'd lean towards "invariants are currently restricted to only read `this`" personally